app/src/main/java/org/mozilla/reference/browser/services/AccountStateMachine.kt

```diff
@@ -0,0 +1,344 @@
+package org.mozilla.reference.browser.services
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.net.Uri
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import mozilla.components.service.fxa.FirefoxAccount
+import mozilla.components.service.fxa.FxaException
+import mozilla.components.service.fxa.Profile
+import mozilla.components.support.base.log.Log
+import org.mozilla.reference.browser.browser.FirefoxAccountsFeature
+import org.mozilla.reference.browser.browser.FirefoxAccountsFeature.Companion.SCOPES
+import org.mozilla.reference.browser.ext.sendToLocalObservers
+import java.util.concurrent.ConcurrentLinkedQueue
+import kotlin.coroutines.CoroutineContext
+
+const val FXA_STATE_PREFS_KEY = "fxaAppState"
+const val FXA_STATE_KEY = "fxaState"
+const val FXA_LAST_SYNCED_KEY = "lastSyncedAt"
+const val FXA_NEVER_SYNCED_TS: Long = 0
+
+fun Context.accountPreferences(): SharedPreferences {
+    return this.getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE)
+}
+
+private fun Context.clearAccountPreferences() {
```

Comment:
nit and general topic: aren't we starting to overuse extension methods? Why not pass the context to this method?